File: languages/php/Bloock/Entity/Key/RsaKeyPair.php
<?php

namespace Bloock\Entity\Key;

use Bloock\GenerateLocalKeyResponse;

/**
 * Represents a rsa key pair, with private and public key.
 */
class RsaKeyPair extends KeyPair
{
    /**
     * Constructs a RsaKeyPair object with the specified parameters.
     * @param string $publicKey
     * @param string $privateKey
     */
    public function __construct(string $publicKey, string $privateKey)
    {
        parent::__construct($publicKey, $privateKey);
    }

    public static function fromProto(GenerateLocalKeyResponse $res): RsaKeyPair
    {
        return new RsaKeyPair($res->getLocalKey()->getKey(), $res->getLocalKey()->getPrivateKey());
    }
}

File: languages/python/bloock/entity/key/rsa_key_pair.py
from bloock._bridge.proto.bloock_keys_pb2 import GenerateLocalKeyResponse
from bloock.entity.key.key_pair import KeyPair


class RsaKeyPair(KeyPair):
    """
    Represents a rsa key pair, with private and public key.
    """
    @staticmethod
    def from_proto(res: GenerateLocalKeyResponse) -> KeyPair:
        return KeyPair(res.local_key.key, res.local_key.private_key)
